### DISCUSSION TITLE
Assertion `engine.getNbBindings() == 2' failed.

### DISCUSSION THREAD

**Discussion by jia0511** [Linaom1214/TensorRT-For-YOLO-Series#33] [Q&A] [answered]
HI ，

我想知道我遇到的这个问题如何解决，我是用cpp进行推理，流程根据官方介绍生成（https://github.com/WongKinYiu/yolov7），
`
wget https://github.com/WongKinYiu/yolov7/releases/download/v0.1/yolov7-tiny.pt
python export.py --weights yolov7-tiny.pt --grid --include-nms
git clone https://github.com/Linaom1214/tensorrt-python.git
python ./tensorrt-python/export.py -o yolov7-tiny.onnx -e yolov7-tiny-nms.trt -p fp16
`
`
blob image
engine.getNbBindings() 3
yolov7: /home/jack/data1/project/tensorrt-python/yolov7/cpp/yolov7.cpp:396: void doInference(nvinfer1::IExecutionContext&, float*, float*, int, cv::Size): Assertion `engine.getNbBindings() == 2' failed.
已放弃 (核心已转储)

`

**Comment by Linaom1214** [comment]
目前C++ 的代码不支持端到端

**Comment by Linaom1214** [comment]
> HI ，
> 
> 我想知道我遇到的这个问题如何解决，我是用cpp进行推理，流程根据官方介绍生成（[https://github.com/WongKinYiu/yolov7），](https://github.com/WongKinYiu/yolov7%EF%BC%89%EF%BC%8C) `wget https://github.com/WongKinYiu/yolov7/releases/download/v0.1/yolov7-tiny.pt python export.py --weights yolov7-tiny.pt --grid --include-nms git clone https://github.com/Linaom1214/tensorrt-python.git python ./tensorrt-python/export.py -o yolov7-tiny.onnx -e yolov7-tiny-nms.trt -p fp16` `blob image engine.getNbBindings() 3 yolov7: /home/jack/data1/project/tensorrt-python/yolov7/cpp/yolov7.cpp:396: void doInference(nvinfer1::IExecutionContext&, float*, float*, int, cv::Size): Assertion`engine.getNbBindings() == 2' failed. 已放弃 (核心已转储)
> 
> `

更新了端到端的支持， enjoy it

**Comment by Linaom1214** [comment, marked as answer]
Now I add the C++ support
https://github.com/Linaom1214/TensorRT-For-YOLO-Series/blob/main/cpp/README.MD
